relayer/src/chain/tx.rs

```diff
@@ -0,0 +1,41 @@
+use core::fmt;
+
+use prost_types::Any;
+
+#[derive(Debug, Clone)]
+pub struct TrackedMsgs {
+    msgs: Vec<Any>,
+    tracking_id: String,
+}
+
+impl TrackedMsgs {
+    pub fn new(msgs: Vec<Any>, tid: &str) -> Self {
+        Self {
+            msgs,
+            tracking_id: tid.into(),
+        }
+    }
+
+    pub fn messages(&self) -> &Vec<Any> {
+        &self.msgs
+    }
+
+    pub fn new_single(msg: Any, tid: &str) -> Self {
+        Self {
+            msgs: vec![msg],
+            tracking_id: tid.into(),
+        }
+    }
+}
+
+impl From<TrackedMsgs> for Vec<Any> {
+    fn from(tm: TrackedMsgs) -> Vec<Any> {
+        tm.msgs
+    }
+}
+
+impl fmt::Display for TrackedMsgs {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "{}; len={}", self.tracking_id, self.msgs.len())
+    }
+}
```

Comment:
I might have abused `Display`, indeed.

> perhaps msgs.tracking_id() and msgs.len() could be used in format args instead of the whole value for better clarity?

Agree. It would be more appropriate to concretely specify which fields we want to display, namely `len` and `tracking_id`.

I added a comment (1d15085e) to at least document the purpose of this type. feel free to replace the use of `Display` with the proposed approach.